contracts/Optimism_SpokePool.sol

```diff
@@ -0,0 +1,80 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.0;
+
+import "@eth-optimism/contracts/libraries/bridge/CrossDomainEnabled.sol";
+import "@eth-optimism/contracts/libraries/constants/Lib_PredeployAddresses.sol";
+import "./SpokePool.sol";
+import "./SpokePoolInterface.sol";
+
+/**
+ * @notice OVM specific SpokePool.
+ * @dev Uses OVM cross-domain-enabled logic for access control.
+ */
+
+contract Optimism_SpokePool is CrossDomainEnabled, SpokePoolInterface, SpokePool {
+    // Address of the L1 contract that acts as the owner of this SpokePool.
+    address public override crossDomainAdmin;
```

Comment:
@chrismaree @mrice32 should `crossDomainAdmin` be in the `SpokePool` or do you think this notion of a cross domain admin will be different for each. L2?

Reply:
Its the contract that calls `relayMessage` on the canonical bridge, so its either the `BridgeAdmin` or the `HubPool`

Reply:
for now `HubPool` me thinks. we might want to change this later but I think lets leave it there optimistically at the moment.